projects/storefrontlib/shared/components/chat-messaging/avatar/avatar.component.ts

```diff
@@ -0,0 +1,23 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ICON_TYPE } from '@spartacus/storefront';
+import { MessageEvent } from '../messaging';
+
+@Component({
+  selector: 'cx-avatar',
+  templateUrl: './avatar.component.html',
+})
+export class AvatarComponent implements OnInit {
+  @Input() event: MessageEvent;
+  iconTypes = ICON_TYPE;
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  getIntitials(name: string) {
+    return name
```

Comment:
What if the input is undefined?

Reply:
adding a check for undefined

Reply:
should we return anything specific instead of null or ' '?

Reply:
if we return an empty string, then we will be having an empty circle as avatar.
Since this information is only for messages created by the customer, then we could return "C" as a customer or better if you find an icon that represents a customer

Reply:
any related icon is not there, let's just show C as these seems like a very rare case rather than adding new icon